src/core/dbus-manager.c

```diff
@@ -1771,11 +1767,37 @@ static int method_unmask_unit_files(sd_bus_message *message, void *userdata, sd_
         return method_disable_unit_files_generic(message, userdata, "enable", unit_file_unmask, error);
 }
 
+static int method_revert_unit_files(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **l = NULL;
+        UnitFileChange *changes = NULL;
+        unsigned n_changes = 0;
+        Manager *m = userdata;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        r = sd_bus_message_read_strv(message, &l);
+        if (r < 0)
+                return r;
+
+        r = bus_verify_manage_unit_files_async(m, message, error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
+
+        r = unit_file_revert(m->unit_file_scope, NULL, l, &changes, &n_changes);
+        if (r < 0)
```

Comment:
shouldn't we call `unit_file_changes_free` here?
This prevents a memory leak. See https://github.com/systemd/systemd/pull/2350/files